### ISSUE TITLE
Middleware doesn't prevent execution main function when file too large

### ISSUE THREAD

**Issue by shazgames1** [richardgirges/express-fileupload#238]
```
router.post(
  "/upload",
  fileUpload({
    limits: { fileSize: 50 * 1024 * 1024 },
    files: 1, 
    abortOnLimit: true,
    limitHandler: (req, res, next) => {
      res.status(400).json({ error: true, message: "Request Entity Too Large" })
    },
  }),
  UploadFile
)
```
I getting errors like ```Cannot set headers after they are sent to the client``` from UploadFile function because limitHandler function has been called and responce already sent to client. How to prevent it or it's bug?

**Comment by PhilHannent** [comment]
I'm not the author but looking at the code you can see that the abortOnLimit is going to send a status, which would conflict with your response and end up with the message you are seeing:

https://github.com/richardgirges/express-fileupload/blob/64587932f7adb10e41702cadade6a873b05e6ca6/lib/processMultipart.js#L75

**Comment by shazgames** [comment]
> I'm not the author but looking at the code you can see that the abortOnLimit is going to send a status, which would conflict with your response and end up with the message you are seeing:
> 
> https://github.com/richardgirges/express-fileupload/blob/64587932f7adb10e41702cadade6a873b05e6ca6/lib/processMultipart.js#L75
> 
> 

So then it should call the next function after error handler function

**Comment by imjordanxd** [comment]
I am also facing this problem. The response code of 413 is being sent. However, the next piece of middleware is being called.

**Comment by sliterok** [comment]
> I am also facing this problem. The response code of 413 is being sent; however, the next piece of middleware is being called. This seems to happen when files being uploaded are very close to the maximum upload size. The file `limit` and `end` event are both being emitted.

I guess the issue is somehow tied to sending files with speeds close to reading from disk. Anyway, should be fixed with PR.

**Comment by imjordanxd** [comment]
any updates on this?

**Comment by israelhuss** [comment]
I'm having the same issue, any solution or workaround?

**Comment by sliterok** [comment]
> I'm having the same issue, any solution or workaround?

Check my PR https://github.com/richardgirges/express-fileupload/pull/254 just above, it should still work

**Comment by RomanBurunkov** [comment]
Fixed in 1.4.2
